library/cpp/include/wavemap/pipeline/pipeline.h

```diff
@@ -26,43 +29,62 @@ class Pipeline {
         thread_pool_(thread_pool ? std::move(thread_pool)
                                  : std::make_shared<ThreadPool>()) {}
 
+  // Copy construction is not supported
+  Pipeline(const Pipeline&) = delete;
+
+  //! Deregister all measurement integrators and map operations
   void clear();
 
+  //! Returns true if an integrator with the given name has been registered
   bool hasIntegrator(const std::string& integrator_name) const;
+  //! Deregister the integrator with the given name. Returns true if it existed.
   bool eraseIntegrator(const std::string& integrator_name);
+  //! Create and register a new integrator
   IntegratorBase* addIntegrator(const std::string& integrator_name,
                                 const param::Value& integrator_params);
+  //! Register the given integrator, transferring ownership
   IntegratorBase* addIntegrator(const std::string& integrator_name,
                                 std::unique_ptr<IntegratorBase> integrator);
+  //! Get a pointer to the given integrator, returns nullptr if it does not
+  //! exist
   IntegratorBase* getIntegrator(const std::string& integrator_name);
+  //! Access all registered integrators (read-only)
   const IntegratorMap& getIntegrators() { return integrators_; }
+  //! Deregister all integrators
   void clearIntegrators() { integrators_.clear(); }
 
+  //! Create and register a new map operation
```

Comment:
Is there any point to having a `removeOperation` similar to the `erase/removeIntegrator` method?

Reply:
Yes, but the operations don't have unique names. So the only interface we could currently support would be to remove them by index. For context, the MeasurementIntegrators are stored in a map with unique (user-provided) names, used by the Inputs to forward their measurement to the right integrator(s). On the other hand, the operations are anonymous and we currently just store them in a vector.

Reply:
Hmm I don't think it would be all that useful then. It doubt there are many cases where people will add and remove operations, or if they do they can just quickly reset and add the desired ones.